### PR TITLE
Create /etc/sysconfig if directory does not exist

### PR DESCRIPTION
**Pull request by TomasKohout**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
Since we are moving from docker to cri-o, I've tried to create correct configuration for Ubuntu OS. When I've specified correct path to irqbalance_config_file which is by the documentation located at /etc/default/irqbalance for Ubuntu, crio failed to start because crio supposed that /etc/sysconfig directory exists. 
#### Special notes for your reviewer:
I'd be glad if someone could point me in the right direction about some unit test for this. :slightly_smiling_face: 
#### Does this PR introduce a user-facing change?
No.

```release-note
none
```

### CODE REVIEW

File: internal/runtimehandlerhooks/high_performance_hooks.go [hunk 1]
```diff
@@ -360,6 +362,14 @@ func RestoreIrqBalanceConfig(irqBalanceConfigFile, irqBannedCPUConfigFile, irqSm
 		return nil
 	}
 	if !fileExists(irqBannedCPUConfigFile) {
+		// check if folder exists, if not create /etc/sysconfig
+		if !fileExists(EtcSysconfig) {
```
Comment (haircommander) [inline internal/runtimehandlerhooks/high_performance_hooks.go:366]: I don't think we can use fileexists, as it checks the path is not a directory (which we expect it to be). Maybe just do
```
if _, err := os.Stat(filename); os.IsNotExist(err) {
...
}
```

**Comment by haircommander** [comment]
thanks for the contribution and for trying out CRI-O @TomasKohout ! unit tests are in `internal/runtimehandlerhooks/high_performance_hooks_test.go` if you want to try to add one. I left one nit
/ok-to-test

File: internal/runtimehandlerhooks/high_performance_hooks.go (hunk 1 above)
Comment (TomasKohout) [reply internal/runtimehandlerhooks/high_performance_hooks.go:366]: You're right. I oversaw the info.IsDir() method call.

**Comment by TomasKohout** [comment]
Hi @haircommander ,

I've added the unit test and changed a bit the logic of getting the directory path (it was ugly for the first time anyway :smile: ). Would you be so kind and check if everything is ok? 

Thank you. :slightly_smiling_face:

**Comment by TomSweeneyRedHat** [comment]
Code changes LGTM, I'm not sure what's going on with the tests, but there is an awful lot of red unhappiness there.

**Comment by TomasKohout** [comment]
Hi @TomSweeneyRedHat ,

yep, I'm also not sure what's wrong. Is Jenkins or some dependent tool ok?

**Comment by haircommander** [comment]
it's defeinitely unrelated to this PR, no to worry

**Comment by haircommander** [comment]
heya @TomasKohout can you rebase your PR to fix the CI issues?

**Comment by haircommander** [comment]
can you drop the merge commit and do a rebase instead? I recommend:

```
git fetch upstream
git rebase upstream/main
git push -f origin main
```

**Comment by TomasKohout** [comment]
It's done. :)

**Comment by haircommander** [comment]
/retitle Create /etc/sysconfig if directory does not exist

finnicky CI :)

**Comment by haircommander** [comment]
/lgtm

thanks for the patience on this one @TomasKohout !

**Comment by TomasKohout** [comment]
Hi @haircommander , NP. :-) 

Should I do a rebase or why do you think the tests are failing? I wan unable to figure it out from provided logs.

**Pull request closed without merging**